packages/devkit/src/project-graph/utils.ts

```diff
@@ -81,3 +82,20 @@ export class ProjectGraphBuilder {
     };
   }
 }
+
+export function isProjectGraphNode(node: any): node is ProjectGraphNode {
+  return (
+    node.name !== undefined &&
+    node.type !== undefined &&
+    node.data !== undefined
+  );
+}
+
+export function isProjectGraphExternalNode(
+  node: any
+): node is ProjectGraphExternalNode {
+  return (
+    isProjectGraphNode(node) &&
+    (node as ProjectGraphExternalNode).data.packageName !== undefined
+  );
+}
```

Comment:
@vsavkin I was unsure whether I should also check the `type === 'npm'` in this function. It would remove some duplicate code from changed `@nrwl/workspace` files.